src/backend/distributed/metadata/metadata_sync.c

```diff
@@ -919,9 +921,18 @@ MarkObjectsDistributedCreateCommand(List *addresses,
 		int forceDelegation = list_nth_int(forceDelegations, currentObjectCounter);
 		List *names = NIL;
 		List *args = NIL;
+		char *objectType = NULL;
 
-		char *objectType = getObjectTypeDescription(address, false);
-		getObjectIdentityParts(address, &names, &args, false);
+		if (IsManagementCommand)
+		{
+			names = namesArg;
+			objectType = "role";
```

Comment:
(e.g., we will probably use this for CREATE DATABASE as well)

Reply:
> (e.g., we will probably use this for CREATE DATABASE as well)

Maybe not because we may want to directly propagate CREATE DATABASE to all nodes as it cannot be executed in a xact block anyway.